src/realm/table.cpp

```diff
@@ -1623,6 +1624,13 @@ std::optional<Mixed> Table::max(ColKey col_key, ObjKey* return_ndx) const
     return AggregateHelper<Table>::max(*this, *this, col_key, return_ndx);
 }
 
+
+StringIndex* Table::get_search_index(ColKey col) const noexcept
+{
+    check_column(col);
+    return dynamic_cast<StringIndex*>(m_index_accessors[col.get_index().val].get());
```

Comment:
The cast suggests that `m_index_accessors` is of type SearchIndex - but it is not. Although I would like it to be.